catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -98,16 +102,32 @@ const editFormSpec: FormSpec<Model.GQLTypes.BucketUpdateInput> = {
     (s) => s.trim() || 'null',
     Types.decode(Types.withFallback(Types.JsonFromString, null)),
   ),
-  fileExtensionsToIndex: R.pipe(
-    R.prop('fileExtensionsToIndex'),
-    Types.decode(Types.fromNullable(IO.string, '')),
-    R.split(','),
-    R.map(R.trim),
-    R.reject((t) => !t),
-    R.uniq,
-    (exts) =>
-      exts.length ? (exts as FP.nonEmptyArray.NonEmptyArray<Types.NonEmptyString>) : null,
-  ),
+  fileExtensionsToIndex: (values) =>
```

Comment:
`values` is untyped (it's just a Record<string, unknown>), but I think it should

In callback `onSubmit(values)` values are untyped too

Reply:
it is untyped bc this is how final-form works. i agree we need a solution for typing (defining) forms properly, and i think it's even backlogged, but right now this set of functions (a "form spec") solves exactly this problem -- converts untyped values dict into a properly typed object or throws an error if that's impossible.